compiler/src/iree/compiler/Codegen/Common/test/propagate_reshapes_by_expansion.mlir

```diff
@@ -86,3 +87,103 @@ func.func @fold_collapse_into_stores_dynamic(%arg0 : tensor<2x?x32xf32>) {
 //       CHECK:   flow.dispatch.tensor.store %{{.+}}, %[[SUBSPAN]]
 //  CHECK-SAME:       offsets = [0, 0, 0], sizes = [2, %[[SHAPE]], 32], strides = [1, 1, 1]
 //  CHECK-SAME:       !flow.dispatch.tensor<writeonly:tensor<2x?x32xf32>>{%[[SHAPE]]}
+
+// -----
+
+#pipeline_layout = #hal.pipeline.layout<constants = 1, bindings = [
+    #hal.pipeline.binding<storage_buffer, Indirect>], flags = Indirect>
+func.func @expand_dest_forall_workgroup_mapped() {
+  %cst = arith.constant 0.000000e+00 : f16
+  %c0 = arith.constant 0 : index
+  %0 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) alignment(64) offset(%c0)
+      flags(Indirect) : !flow.dispatch.tensor<writeonly:tensor<2048x10240xf32>>
+  %1 = tensor.empty() : tensor<2048x10240xf32>
+  %2 = scf.forall (%arg0, %arg1) = (0, 0) to (2048, 10240) step (128, 128)
+    shared_outs(%arg2 = %1) -> (tensor<2048x10240xf32>) {
+    %extracted_slice = tensor.extract_slice %arg2[%arg0, %arg1] [128, 128] [1, 1]
+         : tensor<2048x10240xf32> to tensor<128x128xf32>
+    %3 = tensor.empty() : tensor<8x8x16x8x2xf32>
+    %4 = linalg.fill ins(%cst : f16) outs(%3 : tensor<8x8x16x8x2xf32>) -> tensor<8x8x16x8x2xf32>
+    %5 = tensor.empty() : tensor<8x16x8x8x2xf32>
+    %transposed = linalg.transpose ins(%4 : tensor<8x8x16x8x2xf32>)
+        outs(%5 : tensor<8x16x8x8x2xf32>) permutation = [0, 2, 1, 3, 4]
+    %expanded = tensor.expand_shape %extracted_slice [[0, 1], [2, 3, 4]]
+              output_shape [8, 16, 8, 8, 2] : tensor<128x128xf32> into tensor<8x16x8x8x2xf32>
+    %6 = linalg.copy ins(%transposed : tensor<8x16x8x8x2xf32>)
+         outs(%expanded : tensor<8x16x8x8x2xf32>) -> tensor<8x16x8x8x2xf32>
+    %collapsed = tensor.collapse_shape %6 [[0, 1], [2, 3, 4]] : tensor<8x16x8x8x2xf32> into tensor<128x128xf32>
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %collapsed into %arg2[%arg0, %arg1] [128, 128] [1, 1]
+        : tensor<128x128xf32> into tensor<2048x10240xf32>
+    }
+  } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
```

Comment:
This is a lengthy test. Can you shorten the test by removing all the inner operations except the ones that are necessary for the test?

Reply:
I tried to make the shortest possible test but even removing one of the inner ops causes other patterns/canonicalizations to kick in such that the tested pattern is not even needed.